src/ucp/wireup/wireup.c

```diff
@@ -1226,14 +1226,17 @@ ucp_wireup_check_config_intersect(ucp_ep_h ep, ucp_ep_config_key_t *new_key,
     ucp_ep_config_key_t *old_key;
     ucp_lane_index_t lane, reuse_lane;
     uct_ep_h uct_ep;
+    ucs_status_t status;
+    int old_num, has_config;
 
     *connect_lane_bitmap = UCS_MASK(new_key->num_lanes);
     ucs_queue_head_init(replay_pending_queue);
 
-    if (!ucp_ep_has_cm_lane(ep) ||
-        (ep->cfg_index == UCP_WORKER_CFG_INDEX_NULL)) {
+    has_config = (ep->cfg_index != UCP_WORKER_CFG_INDEX_NULL);
+    if (!ucp_ep_has_cm_lane(ep) || !has_config) {
         /* nothing to intersect with */
-        return;
+        old_num = has_config ? ucp_ep_num_lanes(ep) : UCP_MAX_FAST_PATH_LANES;
+        return ucp_ep_realloc_slow_lanes(ep, old_num, new_key->num_lanes);
```

Comment:
seems every time ucp_ep_realloc_slow_lanes is used, "old_num_lanes" is actually the "current" number of lanes
so maybe no need to pass old_num_lanes, and ucp_ep_realloc_slow_lanes should take it directly from the ep